### PR TITLE
Update Rego policy implementation during Runtime

### LINKED ISSUE #495: Implement update of Rego policy during runtime

### PR DESCRIPTION
**Pull request by oxisto**
This PR introduces a major overhaul of the policy evaluation engine in the assessment service. 

- [x] Restructuring of Rego policy evaluation
- [x] Support for loading policies into a storage backend and retrieving them from the orchestrator in the assessment
- [x] Implementing RPC methods to update metric implementation
- [x] Broadcast metric events to assessment service
- [x] Moving `metrics` from a global variable to the orchestrator `Service`

As a by-product, this introduces several new classes and functions in the `testutils` package. If wanted, I can extract them into a separate PR:

- `NewInMemoryStorage` is a shortcut to quickly create an in-memory storage for unit tests without checking the error value (it will fail the test on error). It also includes handy varargs for storage initialisation
- `StorageWithError` is a struct that can be used to mock a `persistence.Storage` implementation that produces errors on storage operations
- `ListRecvStreamerOf` implements `grpc.ClientStreams` to mock the streaming of a list of messages and then gracefully EOFs the stream.

Fixes #495

### CODE REVIEW

File: persistence/gorm/gorm.go [hunk 1]
```diff
@@ -87,15 +93,17 @@
...
+		&auth.User{},
+		&orchestrator.CloudService{},
+		&assessment.MetricImplementation{},
 	}
-	// Migrate CloudService
-	if err = g.db.AutoMigrate(&orchestrator.CloudService{}); err != nil {
-		err = fmt.Errorf("error during auto-migration: %w", err)
-		return
+
+	for _, typ := range types {
```
Comment (lebogg) [inline persistence/gorm/gorm.go:102]:
```suggestion
	for _, type := range types {
```

File: persistence/gorm/gorm.go [hunk 2]
```diff
@@ -87,15 +93,17 @@
...
+		if err = g.db.AutoMigrate(typ); err != nil {
```
Comment (lebogg) [inline persistence/gorm/gorm.go:103]:
```suggestion
		if err = g.db.AutoMigrate(type); err != nil {
```

File: persistence/gorm/gorm.go [hunk 3]
```diff
@@ -87,15 +93,17 @@
 	}
 
 	// After successful DB initialization, migrate the schema
-	// Migrate User
-	if err = g.db.AutoMigrate(&auth.User{}); err != nil {
-		err = fmt.Errorf("error during auto-migration: %w", err)
-		return
+	var types = []interface{}{
```
Comment (lebogg) [inline persistence/gorm/gorm.go:96]: We could also put everything directly into `AutoMigrate` (variadic function). I can do it in the certificate API PR though.

File: service/assessment/assessment.go [hunk 4]
```diff
@@ -163,6 +175,17 @@
 		o(s)
 	}
 
+	// Default to an in-memory storage, if nothing was explicitly set
```
Comment (lebogg) [inline service/assessment/assessment.go:178]: I think WIthStorage option is missing

**Review by lebogg — APPROVED** [review]
Quick review. Seems good!

File: persistence/gorm/gorm.go (hunk 1 above)
Comment (oxisto) [reply persistence/gorm/gorm.go:102]: `type` is a reserved keyword, that was the reasoning behind `typ`.

File: persistence/gorm/gorm.go (hunk 2 above)
Comment (oxisto) [reply persistence/gorm/gorm.go:103]: See above. Rejected.

File: service/assessment/assessment.go (hunk 4 above)
Comment (oxisto) [reply service/assessment/assessment.go:178]: Ah, this can be removed, this was a remnant from an old implementation attempt.

**Pull request merged**